### DISCUSSION TITLE
[Feature] Manual album arrangement with sections

### DISCUSSION THREAD

**Discussion by mordae** [immich-app/immich#13500] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be nice if Immich allowed for album content to be arranged in a way that facilitates better presentation to other people. Specifically, it would be nice if it was possible to:

1. Reorder photos regardless of their timestamp
   - Selected with 3rd choice in the Album Options
   - Drag & drop to vertical bars between and around the photos to reorder
2. Manually place delimiters to separate sections of album
   - Optionally add text to act as a section heading (otherwise it's just a hard row break)
3. Deterministic addition to album
   - If the last section in album has no label or there are no sections, photos get appended
   - If the last section in album has a label, new section without a label is appended and then the photo is appended

For example (a vacation in Greece):

```
-- Arrival ---------------------------------------------------
+------------+  +----------------------------+  +------------+
| 123        |  | 124                        |  | 125        |
|            |  |                            |  |            |
|            |  |                            |  |            |
|            |  |                            |  |            |
|            |  |                            |  |            |
|            |  |                            |  |            |
|            |  |                            |  |            |
|            |  |                            |  |            |
+------------+  +----------------------------+  +------------+

-- Biking Trip -----------------------------------------------
+------------+  +------------+  +------------+  +------------+
| 433        |  | 126        |  | 127        |  | 131        |
|            |  |            |  |            |  |            |
|            |  |            |  |            |  |            |
|            |  |            |  |            |  |            |
|            |  |            |  |            |  |            |
|            |  |            |  |            |  |            |
|            |  |            |  |            |  |            |
|            |  |            |  |            |  |            |
+------------+  +------------+  +------------+  +------------+
+------------+  +----------------------------+
| 130        |  | 129                        |
|            |  |                            |
|            |  |                            |
|            |  |                            |
|            |  |                            |
|            |  |                            |
|            |  |                            |
|            |  |                            |
+------------+  +----------------------------+


+------------+       <- added later, not yet sorted
| 142        |
|            |
|            |
|            |
|            |
|            |
|            |
|            |
+------------+
```

The simplest data structure required to hold both ordering and delimiters would be `(string | number)[]`, that is:

```
["Arrival", 123, 124, 125, "Biking Trip", 433, 126, 127, 131, 130, 129, "", 142]
```

Possible solutions in DB terms:

1. Manual ordering field would have to be added to the M:N join table between photos and albums and new table with labels would have to be created.
2. Ordering would be saved in a JSON field similar to the example above and any photos related to the album that are not found inside the ordering field would go to a new section at the end of the album when presenting.

Somewhat related to #1689.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#3896
